connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMakerConfig.java

```diff
@@ -200,6 +200,8 @@ public Map<String, String> workerConfig(SourceAndTarget sourceAndTarget) {
 
         // fill in reasonable defaults
         props.putIfAbsent(GROUP_ID_CONFIG, sourceAndTarget.source() + "-mm2");
+        String groupId = props.get(GROUP_ID_CONFIG);
+        props.putIfAbsent(CommonClientConfigs.CLIENT_ID_CONFIG, groupId + "|" + sourceAndTarget);
```

Comment:
We already [include the group ID](https://github.com/apache/kafka/blob/b77b7a6f6f6d39f8064fc5fb11f25b09c58c4228/connect/runtime/src/main/java/org/apache/kafka/connect/util/ConnectUtils.java#L204) when constructing the base client ID from the `DistributedConfig` that will be instantiated with these properties; in most cases, it'll be redundant to add it here.

If it's to cover the case of the shared admin client used for each MM2 herder, we can handle that in [the same way](https://github.com/apache/kafka/blob/b77b7a6f6f6d39f8064fc5fb11f25b09c58c4228/connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java#L77) that we do in vanilla distributed mode in Kafka Connect.

If it's to cover the client ID used by the `DistributedHerder`, we can either leave that as-is for now (I'm not sure it's used for any critical logic/threads/etc. by that class), or implement some way of giving a different client ID specifically to this class without affecting the return value of `ConnectUtils::clientIdBase`.

Reply:
Thanks for the clarification - not going to add the group ID here, but I will still try to set it based on the flow.
In MM2, the group ID only contains the source alias, with specific setups (e.g. star architecture) the group ID is not unique in a single process. DistributedHerder logs do contain the client ID, and I found it useful in many situations (e.g. tracking rebalances/assignments of a specific flow).
Thanks for the hint about the shared admin, will fix that, too.